flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala

```diff
@@ -180,7 +182,37 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem)
     * @return a struct type with the input fieldNames, input fieldTypes, and system fields
     */
   def buildLogicalRowType(tableSchema: TableSchema): RelDataType = {
-    buildLogicalRowType(tableSchema.getFieldNames, tableSchema.getFieldTypes)
+    buildLogicalRowType(tableSchema.getFieldNames, tableSchema.getFieldDataTypes)
+  }
+
+  /**
+   * Creates a struct type with the input fieldNames and input fieldTypes using FlinkTypeFactory
+   *
+   * @param fieldNames field names
+   * @param fieldTypes field types, every element is Flink's [[DataType]]
+   * @return a struct type with the input fieldNames, input fieldTypes, and system fields
+   */
+  def buildLogicalRowType(
+    fieldNames: Array[String],
+    fieldTypes: Array[DataType])
+  : RelDataType = {
+    val logicalRowTypeBuilder = builder
+
+    val fields = fieldNames.zip(fieldTypes)
+    fields.foreach(f => {
+      // time indicators are not nullable
+      val logicalType = f._2.getLogicalType
+      val nullable  = if (FlinkTypeFactory.isTimeIndicatorType(logicalType)) {
+        false
```

Comment:
This is not aligned with the new `CatalogTableSchemaResolver`, should we convert to not null for rowtime in `CatalogTableSchemaResolver#resolve` too? @godfreyhe

Reply:
This fix is only for old planner, and only type info has such problem, because type info does not have nullable attribute.